server/app/services/grid_service_scheduler.rb

```diff
@@ -35,29 +35,37 @@ def instance_count(node_count, instance_count)
   # @param [GridService] grid_service
   # @param [Integer] instance_number
   # @param [Array<HostNode>] nodes
-  # @return [HostNode, NilClass]
+  # @raise [Scheduler::Error]
+  # @return HostNode
   def select_node(grid_service, instance_number, nodes)
     selected_node = nil
     @mutex.synchronize {
       filtered_nodes = self.filter_nodes(grid_service, instance_number, nodes)
       selected_node = self.strategy.find_node(grid_service, instance_number, filtered_nodes)
     }
-    if selected_node
-      node = nodes.find{|n| n == selected_node}
-      node.schedule_counter += 1
+    unless selected_node
+      raise Scheduler::Error, "Strategy #{self.strategy.class.to_s} did not find any node"
     end
 
+    node = nodes.find{|n| n == selected_node}
+    node.schedule_counter += 1
```

Comment:
Not an issue with this PR but we should really get rid of this counter somehow...